merkledag/cidversion.go

```diff
@@ -0,0 +1,26 @@
+package merkledag
+
+import (
+	"fmt"
+)
+
+// CidVersion is a a cid version number that is guaranteed to be valid
+type CidVersion struct {
```

Comment:
> Because nothing prevents me from doing CidVersion{2} either way.

Um, if you do that outside the merkeldag package you will get
```
implicit assignment of unexported field 'version' in merkledag.CidVersion literal
```